### PR TITLE
TileRasterRDD Refactor

### PR DESCRIPTION
**Pull request by jbouffard**
This Pr seeks to add `zoomLevel: Option[Int]` as one of the parameters for the `TileRasterRDD[_]` classes. By doing this, it will be easier to manage each `tileLayer` both in python and in scala.

### CODE REVIEW

File: geopyspark/rdd.py [hunk 1]
```diff
@@ -57,10 +57,20 @@ def __init__(self, geopysc, rdd_type, srdd):
...
+    @property
     def layer_metadata(self):
         """Layer metadata associated with this layer"""
         return json.loads(self.srdd.layerMetadata())
 
+    @property
+    def zoom_level(self):
+        zoom = self.srdd.getZoom()
+
+        if zoom >= 0:
```
Comment (jbouffard) [inline geopyspark/rdd.py]: @echeipesh What do you think of sending a value back to python even when a scala method failed so an `Exception` can be raised on the python side?

**Comment by jbouffard** [comment]
@echeipesh This is ready for review.

File: geopyspark/rdd.py (hunk 1 above)
Comment (echeipesh) [inline geopyspark/rdd.py]: I think its good idea to raise errors on python side as much as possible but in this case we can just return something that gets marshaled to python `None` with less surprise

```scala
  def getZoom: Integer = zoomLevel match {
    case Some(z) => new Integer(z)
    case None => null
  }
```

**Comment by echeipesh** [comment]
Superseded by https://github.com/locationtech-labs/geopyspark/pull/113

**Pull request closed without merging**